pkg/codegen/java/gen.go

```diff
@@ -1515,50 +1534,101 @@ func (mod *modContext) genFunctions(ctx *classFileContext, addClass addClassMeth
 			returnType = ctx.imports.Ref(names.Void)
 		}
 
+		// default method name returns Output<ReturnType>
 		methodName := tokenToFunctionName(fun.Token)
 
+		// another "plain" overload will return CompletableFuture<ReturnType>
+		plainMethodName := methodName + "Plain"
+
 		// Emit datasource inputs method
 		invokeOptions := ctx.ref(names.InvokeOptions)
 
 		if hasAllOptionalInputs(fun) {
+			// Generate invoke that return Output<T>
 			printCommentFunction(ctx, fun, indent)
-			// Add no args invoke
+			// Add no args invoke (returns Output<T>)
 			fprintf(w, "    public static %s<%s> %s() {\n",
-				ctx.ref(names.CompletableFuture), returnType, methodName)
+				ctx.ref(names.Output), returnType, methodName)
 			fprintf(w,
 				"        return %s(%s.Empty, %s.Empty);\n",
 				methodName, argsType, invokeOptions)
 			fprintf(w, "    }\n")
 
+			// Generate invoke that return CompletableFuture<T>
+			printCommentFunction(ctx, fun, indent)
+			fprintf(w, "    public static %s<%s> %s() {\n",
+				ctx.ref(names.CompletableFuture), returnType, plainMethodName)
+			fprintf(w,
+				"        return %s(%s.Empty, %s.Empty);\n",
+				plainMethodName, plainArgsType, invokeOptions)
+			fprintf(w, "    }\n")
 		}
 
-		// Add args only invoke
+		// Output version: add args only invoke
```

Comment:
Maybe gate by `NeedsOutputVersion` like in all other SDKs? 

https://github.com/pulumi/pulumi/blob/ecf65dab2055f1c187d03a2d90aa200167785f76/pkg/codegen/dotnet/gen.go#L1465